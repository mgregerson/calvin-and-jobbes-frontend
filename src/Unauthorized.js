import "./Unauthorized.css";

/** Unauthorized
 *
 * Renders 403 page
 *
 * RoutesList -> Unauthorized
 */

function Unauthorized() {
  console.log("I SWEAR ITS THERE!!!");
  return (
    <div className="Unauthorized">
      <img className="Unauthorized-photo" src="../unauthorized.jpg"></img>
      <h3 className="Unauthorized-message">
        Please login to access this page!!
      </h3>
    </div>
  );
}

export default Unauthorized;
