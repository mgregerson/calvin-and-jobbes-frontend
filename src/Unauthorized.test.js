import { render } from "@testing-library/react";
import Unauthorized from "./Unauthorized";

describe("Unauthorized Component", function () {
  it("renders without crashing", function () {
    render(<Unauthorized />);
  });
  test("matches snapshot", function () {
    const { container } = render(<Unauthorized />);
    expect(container).toMatchSnapshot();
  });
  it("It displays the unauthorized image on the page", function () {
    const { container } = render(<Unauthorized />);
    expect(container.querySelector(".Unauthorized-photo")).toBeInTheDocument();
  });
  it("Displays an unauthorized message", function () {
    const { container } = render(<Unauthorized />);
    expect(container.querySelector(".Unauthorized-message").innerHTML).toEqual(
      "Please login to access this page!"
    );
  });
});
