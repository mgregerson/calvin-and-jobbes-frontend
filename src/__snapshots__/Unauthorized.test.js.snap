// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unauthorized Component matches snapshot 1`] = `
<div>
  <div
    class="Unauthorized"
  >
    <img
      class="Unauthorized-photo"
      src="../unauthorized.jpg"
    />
    <h3
      class="Unauthorized-message"
    >
      Please login to access this page!
    </h3>
  </div>
</div>
`;
